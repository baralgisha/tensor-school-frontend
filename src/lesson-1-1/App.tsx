function App () {
   return (
      <>
         <h1>Топ песен группы "Любэ"</h1>
         <h2>(кто это вообще?)</h2>
         <ul>
            <li><a href="vk.com">vk</a></li>
            <li><a href="vk.com">vk</a></li>
            <li><a href="vk.com">vk</a></li>
         </ul>
      </>
   )
}

export default App