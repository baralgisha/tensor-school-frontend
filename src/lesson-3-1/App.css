html, body, #root {
   height: 100%;
   margin: 0;
}

.flex-container {
   display: flex;
   justify-content: space-between;
   align-items: center;
   width: 700px;
   height: 300px;
}

.flex-container > div {
   width: 100px;
   height: 100px;
   background-color: #DAA520;
   border-radius: 50%;
}

.flex-container :first-child,
.flex-container :last-child {
   align-self: flex-start;
}

.flex-container :nth-child(3) {
   align-self: flex-end;
}
