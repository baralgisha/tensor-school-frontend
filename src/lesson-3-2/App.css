.grid-container {
   display: grid;
   grid-template-columns: repeat(2, 1fr);
   grid-template-rows: 1fr 200px 1fr;
   justify-content: stretch;
   align-items: stretch;
   gap: 20px;
   height: 400px;
   width: 700px;
}

.grid-container > div {
   background-color: #DAA520;
}
