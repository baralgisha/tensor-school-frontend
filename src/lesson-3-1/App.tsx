import './App.css';

function App() {
   return (
   <>
      <section className="flex-container">
         <div></div>
         <div></div>
         <div></div>
         <div></div>
         <div></div>
      </section>
   </>)
}

export default App;