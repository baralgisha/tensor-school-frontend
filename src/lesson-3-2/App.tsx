import './App.css';

function App() {
   return (
   <>
      <section className="grid-container">
         <div></div>
         <div></div>
         <div></div>
         <div></div>
         <div></div>
      </section>
   </>)
}

export default App;