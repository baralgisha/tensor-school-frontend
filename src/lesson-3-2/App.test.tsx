import App from './App';
import {render} from '@testing-library/react';
const { configureToMatchImageSnapshot } = require('jest-image-snapshot');
const puppeteer = require('puppeteer');

const customConfig = { threshold: 0 };
const toMatchImageSnapshot = configureToMatchImageSnapshot({
  customDiffConfig: customConfig,
  noColors: true,
});

let browser;

beforeAll(async () => {
  browser = await puppeteer.launch();
});

expect.extend({ toMatchImageSnapshot });


describe('Сетка', () => {
   it('Верстка не тронута', () => {
      const {container} = render(<App/>);
      expect(container).toMatchSnapshot();
   })

   async function removeBanners(page){
      await page.evaluate(() => {
        (document.querySelectorAll('.flex-container') || []).forEach(el => el.remove());
      });
    }

    it('renders correctly', async () => {
      const page = await browser.newPage();
      await page.goto('http://localhost:3000');

      await removeBanners(page);

      const image = await page.screenshot();

      expect(image).toMatchImageSnapshot();
    });
});