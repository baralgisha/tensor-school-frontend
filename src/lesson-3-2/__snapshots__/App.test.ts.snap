// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Сетка Верстка не тронута 1`] = `
"
        <section class=\\"grid-container\\">
            <div></div>
            <div></div>
            <div></div>
            <div></div>
            <div></div>
        </section>
    

"
`;
